data_model/src/events/data/events.rs

```diff
@@ -170,6 +170,54 @@ mod role {
     }
 }
 
+mod permission_token {
+    //! This module contains `PermissionTokenEvent` and its impls
+
+    use super::*;
+
+    #[derive(
+        Clone,
+        Hash,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Debug,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        Filter,
+    )]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum PermissionTokenEvent {
+        Created(PermissionTokenId),
+        Deleted(PermissionTokenId),
+    }
+
+    impl HasOrigin for PermissionTokenEvent {
+        type Origin = PermissionTokenId;
+
+        fn origin_id(&self) -> &<Self::Origin as Identifiable>::Id {
+            match self {
+                PermissionTokenEvent::Created(id) | PermissionTokenEvent::Deleted(id) => id,
+            }
+        }
+    }
+
+    impl Identifiable for PermissionTokenEvent {
```

Comment:
I don't think we implement `Identifiable` by hand anymore after #2476 was merged. Also, I think it was agreed that events will implement `HashOrigin` instead of `Identifiable`. `HashOrigin` was specifically introduced to handle events